TesseractCore/Exporting.swift

```diff
@@ -1,8 +1,22 @@
 //  Copyright (c) 2015 Rob Rix. All rights reserved.
 
-public func export<T>(graph: Graph<T>) -> String {
-	let edges = join("\n", lazy(graph.edges).map { "\t" + $0.source.identifier.description + " -> " + $0.destination.identifier.description + ";" })
-	return "digraph tesseract {\n\(edges)\n}"
+extension String: Printable {
+    public var description: String {
+        return self
+    }
+}
+
+public func exportDOT<T: Printable>(graph: Graph<T>) -> String {
+    let edges = join("\n", lazy(graph.edges).map { edge in
+        let sourceID = edge.source.identifier
+        let destinationID = edge.destination.identifier
+        let source = graph.nodes[sourceID]
+        let destination = graph.nodes[destinationID]
+        let sourceDescription = source.map { $0.description } ?? ""
+        let destinationDescription = destination.map { $0.description } ?? ""
```

Comment:
Mapping with `toString` instead of the closures will be a little clearer, plus we can remove the extension on String.